src/sentry/event_manager.py

```diff
@@ -1172,6 +1172,11 @@ def materialize_metadata(data, event_type, event_metadata):
     }
 
 
+def get_metadata_for_performance_problem(problem: PerformanceProblem):
+    # TODO make type dynamic, pull it from group type
+    return {"type": "N+1 Query", "value": problem.desc, "title": f"N+1 Query: {problem.desc}"}
+
```

Comment:
i couldn't just use `metadata` on the `event` because each performance issue would have a different metadata

Reply:
I just realised that if we create multiple groups for a job that we're actually modifying the same `metadata` dict for each one. I don't think it's causing problems, but just to be safe let's copy `metadata` before we start modifying it.

Reply:
👍 added that